File: backend/app/models/product.ts
import { DateTime } from 'luxon'
import { BaseModel, belongsTo, column } from '@adonisjs/lucid/orm'
import { BelongsTo } from '@adonisjs/lucid/types/relations'
import Category from './category.js'

export default class Product extends BaseModel {
  @column({ isPrimary: true })
  declare id: number

  @column()
  declare categoryId: number

  @belongsTo(() => Category)
  declare category: BelongsTo<typeof Category>

  @column()
  declare name: string

  @column()
  declare image: string

  @column()
  declare price: number

  @column()
  declare units: number

  @column()
  declare deleted: boolean

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime
}

File: backend/src/models/product.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, FromRow, Serialize, Deserialize)]
pub struct Product {
    pub id: i64,
    pub category_id: i64,
    pub name: String,
    pub image: String,
    pub description: String,
    pub price: f64,
    pub units: i64,
    pub deleted: Option<String>, // 0 false | 1 true
}

#[derive(Debug, FromRow, Serialize, Deserialize)]
pub struct CreateProduct {
    pub category_id: i64,
    pub name: String,
    pub image: String,
    pub description: String,
    pub price: f64,
    pub units: i64,
    pub deleted: Option<String>, // 0 false | 1 true
}
